packages/gatsby-plugin-pixel-facebook/src/pixel/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { Fragment, useMemo } from 'react'
+import type { FC } from 'react'
+import { isServer } from '@vtex/gatsby-theme-store/src/utils/env'
+import { useLazyScript } from '@vtex/gatsby-theme-store/src/sdk/lazyScript/useLazyScript'
+import { usePixelEvent } from '@vtex/gatsby-theme-store/src/sdk/pixel/usePixelEvent'
+import { once } from '@vtex/gatsby-theme-store/src/sdk/once'
+
+import { handler } from './handler'
+
+const setupFBPixel = once((pixelId: string) => {
+  if (isServer) {
+    return
+  }
+
+  let n: any
+
+    /* eslint-disable */
+  ;(function (w) {
+    if (w.fbq) return
+
+    n = w.fbq = function () {
+      n.callMethod ? n.callMethod.apply(n, arguments) : n.queue.push(arguments)
+    }
+    if (!w._fbq) w._fbq = n
+    n.push = n
+    n.loaded = true
+    n.version = '2.0'
+    n.queue = []
+  })(window)
+
+  fbq('init', pixelId)
+  fbq('track', 'PageView')
+})
+
+const useSetupFBPixel = (pixelId: string) =>
+  // We don't need pixelId in hooks deps since setupFBPixel is run once
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  /* eslint-disable */
+  useMemo(() => setupFBPixel(pixelId), [])
+
+interface Props {
+  pixelId: string
+  timeout?: number
+}
+
+const Provider: FC<Props> = ({ children, pixelId, timeout = 2000 }) => {
```

Comment:
Is this timeout too tight?

Reply:
I think so. We usually use 5.5e3 seconds